File: src/main/java/pt/joaomneto/titancompanion/adventure/impl/TPOPAdventure.java
package pt.joaomneto.titancompanion.adventure.impl;


import java.io.BufferedWriter;
import java.io.IOException;

import pt.joaomneto.titancompanion.R;

public class TPOPAdventure extends TWOFMAdventure {

    int copper = 0;

    public TPOPAdventure() {
        super();
        fragmentConfiguration.put(FRAGMENT_VITAL_STATS, new AdventureFragmentRunner(R.string.vitalStats,
                "pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureVitalStatsFragment"));
        fragmentConfiguration.put(FRAGMENT_COMBAT, new AdventureFragmentRunner(R.string.fights,
                "pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureCombatFragment"));
        fragmentConfiguration.put(FRAGMENT_EQUIPMENT, new AdventureFragmentRunner(R.string.goldEquipment,
                "pt.joaomneto.titancompanion.adventurecreation.impl.fragments.tpop.TPOPAdventureEquipmentFragment"));
        fragmentConfiguration.put(FRAGMENT_NOTES, new AdventureFragmentRunner(R.string.notes,
                "pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureNotesFragment"));

    }

    public int getCopper() {
        return copper;
    }

    public void setCopper(int copper) {
        this.copper = copper;
    }

    @Override
    public void storeAdventureSpecificValuesInFile(BufferedWriter bw) throws IOException {

        bw.write("copper=" + copper + "\n");
    }

    @Override
    protected void loadAdventureSpecificValuesFromFile() {
        String copper = getSavedGame().getProperty("copper");

        this.copper = Integer.valueOf(copper!=null && copper.length()>0?copper:"0");
    }

}

File: src/main/java/pt/joaomneto/titancompanion/adventure/impl/TPOPAdventure.kt
package pt.joaomneto.titancompanion.adventure.impl

import pt.joaomneto.titancompanion.R
import pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureCombatFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureNotesFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureVitalStatsFragment
import pt.joaomneto.titancompanion.adventurecreation.impl.fragments.tpop.TPOPAdventureEquipmentFragment
import pt.joaomneto.titancompanion.util.AdventureFragmentRunner
import java.io.BufferedWriter
import java.io.IOException

class TPOPAdventure : TWOFMAdventure(
    arrayOf(
        AdventureFragmentRunner(
            R.string.vitalStats,
            AdventureVitalStatsFragment::class
        ),
        AdventureFragmentRunner(
            R.string.fights,
            AdventureCombatFragment::class
        ),
        AdventureFragmentRunner(
            R.string.goldEquipment,
            TPOPAdventureEquipmentFragment::class
        ),
        AdventureFragmentRunner(
            R.string.notes,
            AdventureNotesFragment::class
        )
    )
) {

    var copper = 0

    @Throws(IOException::class)
    override fun storeAdventureSpecificValuesInFile(bw: BufferedWriter) {
        bw.write("standardPotion=$standardPotion\n")
        bw.write("standardPotionValue=$standardPotionValue\n")
        bw.write("copper=$copper\n")
        bw.write("gold=$gold\n")
    }

    override fun loadAdventureSpecificValuesFromFile() {
        val copperS = savedGame.getProperty("copper")
        val goldS = savedGame.getProperty("gold")
        val standardPotionS = savedGame.getProperty("standardPotion")
        val standardPotionValueS = savedGame.getProperty("standardPotionValue")

        this.copper = if (copperS.isNullOrBlank()) 0 else Integer.parseInt(copperS)
        this.gold = if (copperS.isNullOrBlank()) 0 else Integer.parseInt(goldS)
        this.standardPotion = if (standardPotionS.isNullOrBlank()) -1 else Integer.parseInt(standardPotionS)
        this.standardPotionValue = if (standardPotionValueS.isNullOrBlank()) -1 else Integer.parseInt(standardPotionValueS)
    }
}
